Microsoft compiler crashes building tts_windows.cpp

### Tested versions

Latest master (1f012286cf9321b2209259539d1e638dac4eed73).

### System information

Windows 10 (build 19045) - Multi-window, 2 monitors - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3080 (NVIDIA; 32.0.15.8115) - AMD Ryzen 7 3700X 8-Core Processor (16 threads) - 31.95 GiB memory

### Issue description

I can't seem to get master to build in the Visual Studio 2019.  It always crashes on tts_windows.cpp.

```
**********************************************************************
** Visual Studio 2019 Developer Command Prompt v16.11.53
** Copyright (c) 2021 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x64'

C:\Program Files (x86)\Microsoft Visual Studio\2019\Community>e:

E:\>cd projects\godot\godot_engine_tremble

E:\Projects\godot\godot_engine_tremble>scons p=windows tools=yes debug_symbols=yes target=editor profiler=tracy profiler_path=E:\projects\godot\godot_engine_tremble\modules\godot_tracy\tracy -j1
scons: Reading SConscript files ...
Building for platform "windows", architecture "x86_64", target "editor".
scons: done reading SConscript files.
scons: Building targets ...
[  0%] Copy("bin\D3D12Core.dll", "C:\Users\jitspoe\AppData\Local\Godot\build_deps\agility_sdk\build\native\bin\x64\D3D12Core.dll")
[  0%] Copy("bin\d3d12SDKLayers.dll", "C:\Users\jitspoe\AppData\Local\Godot\build_deps\agility_sdk\build\native\bin\x64\d3d12SDKLayers.dll")
[  3%] Compiling platform\windows\tts_windows.cpp ...
C:\Program Files (x86)\Windows Kits\10\include\10.0.26100.0\cppwinrt\winrt/base.h(5069): fatal error C1001: Internal compiler error.
(compiler file 'msc1.cpp', line 1603)
 To work around this problem, try simplifying or changing the program near the locations listed above.
If possible please provide a repro here: https://developercommunity.visualstudio.com
Please choose the Technical Support command on the Visual C++
 Help menu, or open the Technical Support help file for more information
INTERNAL COMPILER ERROR in 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\cl.exe'
    Please choose the Technical Support command on the Visual C++
    Help menu, or open the Technical Support help file for more information
scons: *** [bin\obj\platform\windows\tts_windows.windows.editor.x86_64.obj] Error 3221225477
scons: building terminated because of errors.
INFO: Time elapsed: 00:16:03.39
```

<img width="352" height="200" alt="Image" src="https://github.com/user-attachments/assets/72ec863a-2221-41c5-814e-d0924814abf3" />

Compiler callstack, if that's useful:

```
>	c1xx.dll!DoArgumentDependentLookup()	Unknown
 	c1xx.dll!function_call_action()	Unknown
 	c1xx.dll!PrimaryParser::Parse()	Unknown
 	c1xx.dll!`CallPrimaryParser'::`2'::<lambda_1>::operator()()	Unknown
 	c1xx.dll!CallPrimaryParser()	Unknown
 	c1xx.dll!main_compile()	Unknown
 	c1xx.dll!Trap_main_compile()	Unknown
 	c1xx.dll!CallMain()	Unknown
 	c1xx.dll!Trap_CallMain()	Unknown
 	c1xx.dll!InvokeCompilerPassW()	Unknown
 	cl.exe!execute()	Unknown
 	cl.exe!dopass()	Unknown
 	cl.exe!passes(enum pass_type const *,struct context_t *,int,int (*)(struct context_t *))	Unknown
 	cl.exe!compile_worklist()	Unknown
 	cl.exe!compile()	Unknown
 	cl.exe!wmain()	Unknown
 	cl.exe!__scrt_common_main_seh()	Unknown
 	kernel32.dll!BaseThreadInitThunk()	Unknown
 	ntdll.dll!RtlUserThreadStart()	Unknown
```

```asm
c1xx.dll!DoArgumentDependentLookup(void):
00007FFCA9A1D29C  mov         qword ptr [rsp+10h],rbx  
00007FFCA9A1D2A1  mov         qword ptr [rsp+18h],rsi  
00007FFCA9A1D2A6  mov         qword ptr [rsp+20h],rdi  
00007FFCA9A1D2AB  push        rbp  
00007FFCA9A1D2AC  push        r14  
00007FFCA9A1D2AE  push        r15  
00007FFCA9A1D2B0  mov         rbp,rsp  
00007FFCA9A1D2B3  sub         rsp,80h  
00007FFCA9A1D2BA  mov         rdi,rdx  
00007FFCA9A1D2BD  mov         rbx,rcx  
00007FFCA9A1D2C0  movzx       eax,word ptr [rcx+14h]  
00007FFCA9A1D2C4  test        byte ptr [rcx+19h],2  
00007FFCA9A1D2C8  jne         DoArgumentDependentLookup+3B6h (07FFCA9A1D652h)  
00007FFCA9A1D2CE  mov         ecx,189h  
00007FFCA9A1D2D3  cmp         ax,cx  
00007FFCA9A1D2D6  je          DoArgumentDependentLookup+3CBh (07FFCA9A1D667h)  
00007FFCA9A1D2DC  cmp         ax,13h  
00007FFCA9A1D2E0  je          DoArgumentDependentLookup+3CBh (07FFCA9A1D667h)  
00007FFCA9A1D2E6  lea         r15d,[rcx-3Ah]  
00007FFCA9A1D2EA  cmp         ax,r15w  
00007FFCA9A1D2EE  je          `Module::InterfaceReader::collections'::`2'::`dynamic atexit destructor for 'collections''+7F1CAh (07FFCA9BF03FAh)  
00007FFCA9A1D2F4  mov         rcx,rbx  
00007FFCA9A1D2F7  test        byte ptr [rcx+1Dh],10h  
00007FFCA9A1D2FB  jne         DoArgumentDependentLookup+2F1h (07FFCA9A1D58Dh)  
00007FFCA9A1D301  mov         rax,qword ptr [rcx+20h]  
>00007FFCA9A1D305  cmp         byte ptr [rax+11h],16h  < Crashes here
00007FFCA9A1D309  je          DoArgumentDependentLookup+2F1h (07FFCA9A1D58Dh)  
00007FFCA9A1D30F  test        byte ptr [rcx+17h],10h  
00007FFCA9A1D313  jne         DoArgumentDependentLookup+3CBh (07FFCA9A1D667h)  
00007FFCA9A1D319  mov         rax,qword ptr [rcx+20h]  
00007FFCA9A1D31D  mov         rsi,qword ptr [rax]  
00007FFCA9A1D320  cmp         rsi,qword ptr [PchS+2208h (07FFCAA1ABC88h)]  
00007FFCA9A1D327  je          DoArgumentDependentLookup+3CBh (07FFCA9A1D667h)  
00007FFCA9A1D32D  and         qword ptr [rbp-28h],0  
00007FFCA9A1D332  mov         rax,qword ptr [HeapManager::ActiveHeaps+160h (07FFCAA1B69D0h)]  
00007FFCA9A1D339  mov         rcx,qword ptr [HeapManager::ActiveHeaps+158h (07FFCAA1B69C8h)]  
00007FFCA9A1D340  sub         rax,rcx  
00007FFCA9A1D343  mov         r14d,28h  
00007FFCA9A1D349  cmp         rax,r14  
00007FFCA9A1D34C  jb          DoArgumentDependentLookup+446h (07FFCA9A1D6E2h)  
00007FFCA9A1D352  lea         rax,[rcx+28h]  
00007FFCA9A1D356  mov         qword ptr [HeapManager::ActiveHeaps+158h (07FFCAA1B69C8h)],rax  
00007FFCA9A1D35D  mov         qword ptr [rcx],rcx  
00007FFCA9A1D360  mov         qword ptr [rcx+8],rcx  
00007FFCA9A1D364  mov         qword ptr [rcx+10h],rcx  
00007FFCA9A1D368  mov         word ptr [rcx+18h],101h  
00007FFCA9A1D36E  mov         qword ptr [rbp-30h],rcx  
```


### Steps to reproduce

Attempt to compile master with command prompt for VS 2019 using this command line: `scons p=windows tools=yes debug_symbols=yes target=editor -j1`

### Minimal reproduction project (MRP)

N/A

> 10.0.26100.0\cppwinrt\winrt/base.h(5069): fatal error C1001: Internal compiler error.

Try updating both Visual Studio (current version is 16.11.55) and SDK. But it's compiler error in the Windows SDK header, so probably should be reported to https://developercommunity.visualstudio.com

--- a/platform/windows/detect.py
+++ b/platform/windows/detect.py
@@ -128,15 +128,66 @@ def detect_build_env_arch():
     return ""
 
 
+def check_mssdk_version(env, version, msvc_ver):
+    from SCons.Tool.MSCommon.MSVC.WinSDK import get_msvc_sdk_version_list
+
+    def int_or_zero(i):
+        try:
+            return int(i)
+        except (TypeError, ValueError):
+            return 0
+
+    def ver_parse(a):
+        return [int_or_zero(i) for i in a.split(".")]
+
+    mssdk = version
+    mssdk_list = get_msvc_sdk_version_list(msvc_ver, False)
+    if not mssdk_list:
+        return ""
+
+    if mssdk == "":
+        mssdk = mssdk_list[0]
+    if mssdk not in mssdk_list:
+        print_error(f"Specified Windows SDK version {mssdk} is not installed, installed versions are: {mssdk_list}.")
+        sys.exit(255)
+
+    if env["winrt"]:
+        if msvc_ver == "14.2":  # Visual Studio 2019, force supported SDK.
+            if "10.0.22621.0" in mssdk_list:
+                mssdk = "10.0.22621.0"
+            else:
+                print_error(
+                    "Visual Studio 2019 require Windows SDK 10.0.22621.0 to use WinRT, install supported Windows SDK version "
+                    'or disable WinRT support by passing "winrt=no" to the SCons command line.'
+                )
+                sys.exit(255)
+        elif ver_parse(mssdk)[2] < 22621:
+            print_error(
+                "Windows SDK 10.0.22621.0 or newer is required to use WinRT, install supported Windows SDK version "
+                'or disable WinRT support by passing "winrt=no" to the SCons command line.'
+            )
+            sys.exit(255)
+
+    print(f"Using Visual Studio {msvc_ver} with Windows SDK {mssdk}.")
+    return mssdk
+
+
 def get_tools(env: "SConsEnvironment"):
     from SCons.Tool.MSCommon import msvc_exists
+    from SCons.Tool.MSCommon.vc import get_default_version
 
     if os.name != "nt" or env.get("use_mingw") or not msvc_exists():
         return ["mingw"]
     else:
         msvc_arch_aliases = {"x86_32": "x86", "arm32": "arm"}
+        msvc_ver = get_default_version(env)
+        if env.get("msvc_version"):
+            msvc_ver = env.get("msvc_version")
+
         env["TARGET_ARCH"] = msvc_arch_aliases.get(env["arch"], env["arch"])
         env["MSVC_VERSION"] = env["MSVS_VERSION"] = env.get("msvc_version")
+        env["MSVC_SDK_VERSION"] = check_mssdk_version(env, env.get("mssdk_version"), msvc_ver)
+
         return ["msvc", "mslink", "mslib"]
 
 
@@ -165,6 +216,7 @@ def get_opts():
         ("mingw_prefix", "MinGW prefix", mingw),
         EnumVariable("windows_subsystem", "Windows subsystem", "gui", ["gui", "console"], ignorecase=2),
         ("msvc_version", "MSVC version to use. Handled automatically by SCons if omitted.", ""),
+        ("mssdk_version", "Windows SDK version to use. Handled automatically by SCons if omitted.", ""),
         BoolVariable("use_mingw", "Use the Mingw compiler, even if MSVC is installed.", False),
         BoolVariable("use_llvm", "Use the LLVM compiler", False),
         BoolVariable("use_static_cpp", "Link MinGW/MSVC C++ runtime libraries statically", True),

